openmdao/core/component.py

```diff
@@ -352,6 +352,11 @@ def _setup_partials(self):
             of, wrt = key
             self._declare_partials(of, wrt, dct)
 
+        if self.matrix_free and self._subjacs_info:
+            simple_warning(f"{self.msginfo}: matrix free component has declared the following "
+                           f"partials: {sorted(self._subjacs_info)}, which will allocate "
```

Comment:
Warning message isn't tested (though I don't remember if we test warnings rigorously.)